meson.build

```diff
@@ -46,8 +46,21 @@ if get_option('buildtype') in ['debug', 'debugoptimized']
     add_project_arguments('-DNANOARROW_DEBUG', language: 'cpp')
 endif
 
+if host_machine.system() == 'windows'
+    # On Windows the TestJSONComparison class cannot be exported to shared libs
+    # properly because it contains stl class members. As a workaround, we force
+    # static libraries for now. See also
+    # https://github.com/apache/arrow-nanoarrow/pull/776#issuecomment-2962559614
+    target_type = 'static_library'
+else
+    target_type = 'library'
+endif
+
 nanoarrow_dep_args = []
-if host_machine.system() == 'windows' and get_option('default_library') == 'shared'
+if (host_machine.system() == 'windows'
```

Comment:
The formatting enforced by meson-format here is a bit unfortunate, but will be fixed when version 1.9.0 gets released